smarts/core/utils/traffic_history_serivce.py

```diff
@@ -77,6 +80,17 @@ def __init__(self, history_file_path):
         self._prepare_next_batch()
         self._receive_data_conn.recv()
 
+    def teardown(self):
+        if self.is_in_use:
+            self._request_queue.put(Traffic_history_service.QueueDone)
+            self._request_queue.close()
+            self._request_queue = None
+            self._fetch_history_proc.join(timeout=3)
```

Comment:
Here you might check `isAlive()` and, if the join timed out, print some sort of warning message telling the user (in case they want to kill the process manually).  (imo, not that big of a deal though, either way.)